### PR TITLE
[Merged by Bors] - feat: port Topology.Order

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by rwbarton** [comment]
Is this still WIP?

**Comment by Ruben-VandeVelde** [comment]
Ref https://github.com/leanprover-community/mathlib/pull/18312

**Comment by rwbarton** [comment]
I notice that `t.IsOpen` seems to work fine in Lean 4 even now that `TopologicalSpace` is a class.
Is there a reason you avoid it in favor of `@IsOpen _ t`?

**Comment by rwbarton** [comment]
Hmm, I guess because that refers to `TopologicalSpace.IsOpen` and not `IsOpen`?
Is there some way we can unify these? It's a little annoying that the shorter/more natural form is not the preferred one.

**Comment by urkud** [comment]
Most of the time we use `IsOpen` without explicit topology. Let's add notation `IsOpen[t] s`.

File: Mathlib/Topology/Basic.lean [hunk 1]
```diff
@@ -1569,6 +1569,10 @@ structure Continuous (f : α → β) : Prop where
   is_open_preimage : ∀ s, IsOpen s → IsOpen (f ⁻¹' s)
 #align continuous Continuous
 
+set_option quotPrecheck false in
```
Comment (ChrisHughes24) [inline Mathlib/Topology/Basic.lean:1572]: This needs a docstring

**Review by ChrisHughes24 — APPROVED** [review]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**